common/beerocks/bwl/shared/nl_genl_socket.cpp

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "nl_genl_socket.h"
+
+#include <linux/netlink.h>
+
+namespace bwl {
+
+nl_genl_socket::nl_genl_socket() : netlink_socket(NETLINK_GENERIC) {}
+
+nl_genl_socket::~nl_genl_socket() {}
```

Comment:
I personally prefer to put such empty or very simple function defintions inline, so the cpp file can be removed entirely.

Note that putting stuff inline like that not only saves code size and performance, but sometimes even compile time because the compiler can elide the call right away and simplify its intermediate representation. (In a previous life, I did research on compiler optimisations.)

Reply:
Done